climada/hazard/landslide.py

```diff
@@ -188,304 +127,170 @@ def __init__(self):
         self.tag.haz_type = 'LS'
 
 
-    def _get_window_from_coords(self, path_sourcefile, bbox):
-        ###### would fit better into base calss for sub-function of hazard.set_raster()########
-        """get row, column, width and height required for rasterio window function
-        from coordinate values of bounding box
+    def set_ls_hist(self, bbox, path_sourcefile, res=0.0083333, check_plots=False):
+        """
+        Set historic landslide (ls) raster hazard from historical point records,
+        for example as can be retrieved from the NASA COOLR initiative,
+        which is the largest global ls repository, for a specific geographic
+        extent.
+        Points are assigned to the gridcell they fall into, and the whole grid-
+        cell hence counts as equally affected.
+        Event frequencies are roughly estimated by 1/(total time passed in event set),
+        but such estimates are not meaningful and shouldn't be used for 
+        probabilistic calculations! Use the probabilistic method for this!
+        
+        See tutorial for details; the global ls catalog from NASA COOLR can be
+        downloaded from https://maps.nccs.nasa.gov/arcgis/apps/webappviewer/index.html?id=824ea5864ec8423fb985b33ee6bc05b7
+
         Parameters:
-            path_sourcefile (str or Path): path of file from which window should be read in
-            bbox (array): [north, east, south, west]
+        ----------
+        bbox : tuple
+            (minx, miny, maxx, maxy) geographic extent of interest
+         path_sourcefile : str
+             path to shapefile (.shp) with ls point data
+        res : float
+            resolution in degrees of the final grid cells which are created.
+            (1° ~ 111 km at the equator). 
+            Affects the hazard extent! Default is 0.00833
```

Comment:
espg 4326

Reply:
I dont see how this is related. 1° at the equator is always ca 111km, irrespective of the projection, as long as a full circle is 360°.
And of course, one can manually reproject the gdf if one wanted to, for some reason.

Reply:
Ok, then I probably misunderstood something. My concern is the following: with this method, your are dividing the surface in square cells of side lenght defined in ° (the raster). Then, you assign the landslide events to each raster cell. In this case, depending on the projection used, the cells and associated landslide events will vastly differ no?

Reply:
correct. i believe this is the principle of a raster file.

Reply:
I don't think so. The idea of a raster file is precisely the opposite. The information included in the raster allows to work with a regular raster on a mathematical level, but there is an implicit projection. Hence, the raster cell size is not independent on the projection. If instead of using degrees, one uses kilometers, the raster file still has the same mathematical regular grid representation, but the square cell side length is completely different.

Reply:
so, i now added in commit 1d0ca7a  the possibility to put in a gdf or a string of where the gdf is located, such that one may re-project the gdf into a more useful projection beforehand. also added some lengthy Note section in the docstring to explain how resolution affects the hazard extent

Reply:
Did you add tests for both use cases?

Reply:
yep. that's the test i still wanted to add!